### ISSUE TITLE
Plural keys are not sorted properly (they appear at bottom)

### ISSUE THREAD

**Issue by hgezim** [i18next/i18next-parser#169]
- Is this a bug or a feature request?
Bug

- What version are you running (`0.x` or `1.x`)? 
1.0.0-beta34

- Which runner (cli, gulp...)?
cli

## What I'm trying to do?
Sort my keys.

## How am I doing it?
By setting `sort` to true in config.

Here is my config file:

```
// i18next-parser.config.js

module.exports = {  
    defaultNamespace: 'common',
    // Default namespace used in your i18next config
  
    defaultValue: '',
    // Default value to give to empty keys
  
    indentation: 4,
    // Indentation of the catalog files
    
    // see below for more details
    lexers: {
      js: ['JavascriptLexer'], // if you're writing jsx inside .js files, change this to JsxLexer
      jsx: ['JsxLexer'],
  
      ts: ['TypescriptLexer'],
      tsx: ['TypescriptLexer'],
  
      default: ['JavascriptLexer']
    },
  
    lineEnding: 'auto',
    // Control the line ending. See options at https://github.com/ryanve/eol
  
    locales: ['en', 'de', 'cs'],
    // An array of the locales in your applications
  
    namespaceSeparator: ':',
    // Namespace separator used in your translation keys
    // If you want to use plain english keys, separators such as `.` and `:` will conflict. You might want to set `keySeparator: false` and `namespaceSeparator: false`. That way, `t('Status: Loading...')` will not think that there are a namespace and three separator dots for instance.
  
    output: 'static/locales/$LOCALE/$NAMESPACE.json',
    // Supports $LOCALE and $NAMESPACE injection
    // Supports JSON (.json) and YAML (.yml) file formats
    // Where to write the locale files relative to process.cwd()
  
    input: undefined,
    // An array of globs that describe where to look for source files
    // relative to the location of the configuration file
  
    reactNamespace: false,
    // For react file, extract the defaultNamespace - https://react.i18next.com/components/translate-hoc.html
    // Ignored when parsing a `.jsx` file and namespace is extracted from that file.
  
    sort: true,
    // Whether or not to sort the catalog
  
    useKeysAsDefaultValue: false,
    // Whether to use the keys as the default value; ex. "Hello": "Hello", "World": "World"
    // The option `defaultValue` will not work if this is set to true
  
    verbose: false
    // Display info about the parsing including some stats
  }
```

The command I'm running:

```
i18next  --config i18next-parser.config.js components/**/*.{ts,tsx} pages/**/*.{ts,tsx} server/**/*.{ts,tsx} services/**/*.{ts,tsx} store/**/*.{ts,tsx} types/**/*.{ts,tsx} util/**/*.{ts,tsx}
```

## What was the result?

`RatingsWithCount` has a bunch of keys after it and then we see `RatingsWithCount_plural`.

```
{
    ...,
    "RatingsWithCount": "{{count}} Rating",
    "RecipeDoesNotExist": "Recipe does not exist",
    "RecipeName": "{{recipeName}} recipe",
    "Save": "Save",
    "SearchForRecipes": "Search for recipes",
    "SearchResultsFor": "Results for: {{searchTerm}}",
    "SearchTitle": "{{recipeType}} recipes",
    "StoreDisabled": "This store has been disabled.",
    "ThisIsTheEnd": "You have reached the end.",
    "UserRating": "Your rating: {{ratingNumber}} of 5",
    "VisitStore": "Visit Store",
    "RatingsWithCount_plural": "{{count}} Ratings"
}
```
## What was I expecting?

`RatingsWithCount` should be followed by `RatingsWithCount_plural`.

```
{
    ...,
    "RatingsWithCount": "{{count}} Rating",
    "RatingsWithCount_plural": "{{count}} Ratings",
    "RecipeDoesNotExist": "Recipe does not exist",
    "RecipeName": "{{recipeName}} recipe",
    "Save": "Save",
    "SearchForRecipes": "Search for recipes",
    "SearchResultsFor": "Results for: {{searchTerm}}",
    "SearchTitle": "{{recipeType}} recipes",
    "StoreDisabled": "This store has been disabled.",
    "ThisIsTheEnd": "You have reached the end.",
    "UserRating": "Your rating: {{ratingNumber}} of 5",
    "VisitStore": "Visit Store"
}

**Comment by karellm** [comment]
I tried to duplicate and write some failing tests cases without success. Can you please provide a failing test case or a git repo where I can reproduce this?

**Comment by vincaslt** [comment]
I haven't checked how it behaves with the current version, but I don't see this effect in #191 and it changes a how plurals are handled, so try the next version when it's merged.

**Comment by karellm** [comment]
I never got around to finish this. If I remember correctly, if you have a file with existing catalog with `_plural` keys in it, they are not taken into account for the sorting. I doubt that the new PR will fix that. I wanted to find a solution a little more elegant than looping through the entire catalog again to sort it.

**Comment by vincaslt** [comment]
I tried it both with existing and new catalogs, but maybe you're right, I haven't tested this extensively.

**Comment by trew** [comment]
May I suggest that the catalog is sorted at the end instead?

Something like this works fine:
https://github.com/trew/i18next-parser/commit/e8140a431ee66de86316242edc5ed8a6d3f866d3

It breaks the provided sort-function, but I don't think it was added to support sorting by value anyway.

**Comment by karellm** [comment]
@trew Could you submit a PR? I will review it for sure in the upcoming days.

**Comment by karellm** [comment]
This is now fixed in `5.0.0` thanks to @Elindorath 's great work. Please read the [Changelog](https://github.com/i18next/i18next-parser/blob/master/docs/migration.md) and [migration guide](https://github.com/i18next/i18next-parser/blob/master/docs/migration.md) before updating.
